modules/rte_mis_student_tracking/src/Commands/StudentTrackingCommands.php

```diff
@@ -0,0 +1,372 @@
+      }
+      else {
+        $nids = $this->entityTypeManager->getStorage('mini_node')->getQuery()
+          ->accessCheck(FALSE)
+          ->condition('type', 'allocation')
+          ->condition('status', 1)
+          ->condition('field_student_allocation_status', $from_state)
+          ->execute();
+      }
+      $user_storage = $this->entityTypeManager->getStorage('user');
+      // Get a user with app admin role so that state transition does
+      // not fail.
+      $user_ids = $user_storage->getQuery()
+        ->accessCheck(FALSE)
+        ->condition('roles', 'app_admin')
+        ->condition('status', 1)
+        ->execute();
+      // Load allocation nodes and update status.
+      foreach ($nids as $nid) {
+        $mini_node = $this->entityTypeManager->getStorage('mini_node')->load($nid);
+        if ($mini_node instanceof EckEntityInterface && $mini_node->bundle() == 'allocation') {
+          // Get the current state.
+          $current_state = workflow_node_current_state($mini_node, 'field_student_allocation_status');
+          if ($current_state == $from_state) {
+            $transition = WorkflowTransition::create([
+              0 => $current_state,
+              'field_name' => 'field_student_allocation_status',
+            ]);
+            // Set the target entity.
+            $transition->setTargetEntity($mini_node);
+            // Set the target state with require details.
+            $transition->setValues($to_state, reset($user_ids), $this->time->getRequestTime(), $this->t('Updated'));
+            // Execute the transition and update the allocation entity.
+            $transition->executeAndUpdateEntity();
+          }
+        }
+        else {
+          $this->output()->writeln("Could not update allocation mini node.");
+        }
+      }
+
+      $this->output()->writeln('Allocation update process has been completed.');
+    }
+    else {
+      // If user declines, output a message.
+      $this->output()->writeln('Update operation cancelled.');
+    }
+  }
+
+  /**
+   * Command to create bulk data for student performance mini node.
+   *
+   * @param int $count
+   *   Number of mini nodes to generate.
+   *
+   * @command rte_mis_student_tracking:bulk-generate-student-peformance
+   * @aliases bgsp
+   * @usage bgsf 1000
+   *   This will create 1000 mini nodes for student performance.
+   */
+  public function bulkGenerateStudentPerformance($count = 100) {
+    $mini_node_storage = $this->entityTypeManager->getStorage('mini_node');
+    // Get max 5 students.
+    $student_ids = $mini_node_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', 'student_details')
+      ->condition('status', 1)
+      ->range(0, 5)
+      ->execute();
+
+    // Get max 5 schools.
+    $school_ids = $mini_node_storage->getQuery()
+      ->accessCheck(FALSE)
+      ->condition('type', 'school_details')
+      ->condition('status', 1)
+      ->range(0, 5)
+      ->execute();
```

Comment:
Are we generating test data here?

Reply:
Yes @kunal-sin, generating dummy data for testing.

Reply:
Then pls update the function name and include test in it.

Reply:
Update the class name itself as all the functions here are used to update/create data for testing only.